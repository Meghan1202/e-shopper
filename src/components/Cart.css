  th{
    border: 1px solid black;
    background-color: darkgray;
  }

  td{
    border: 1px solid black;
    background-color: lightgrey;
  }
  
  .Cart__table {
    table-layout: auto;
    width: 100%;  
    border-collapse: collapse;
    border: 1px solid black;
  }
  
  .Total__Value {
    margin-top: auto;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    transition: 0.3s;
  }
  
  .card:hover {
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
  }

  .Total__Value__containerr {
    padding: 2px 16px;
  }

  #Checkout__button{
    width: fit-content;
  }