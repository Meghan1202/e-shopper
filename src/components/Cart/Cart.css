  
  .Cart__Body{
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    min-height: 90vh;
    justify-content: space-between;
  }

  .Cart__Container{
    display: flex;
    flex-direction: column;
  }

  .Checkout__And__Continue__Container {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    padding: 0.62rem;
  }

  .Total__Value {
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    transition: 0.3s;
    width:  21rem;
    height: 12rem;
    border-radius: 6px;
    background-image: linear-gradient(rgb(245, 245, 245), rgb(207, 207, 207));
  }

  .Shop__button {
    width: fit-content;
    height: 35px;
    border-radius: 6px;
  }

  .Value__Container{
    height: 7.5rem;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding-left: 0.62rem;
    padding-right: 1.56rem;
  }
  
  .Checkout__Card:hover {
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
  }

  .Total__Value__container {
    padding: 2px 16px;
    display: flex;
    flex-direction: row-reverse;
  }

  .Checkout__button{
    width: fit-content;
    height: 35px;
    width: 100px;
    border-radius: 6px;
    background-image: linear-gradient(yellow, rgb(255, 187, 0));
  }


@media (max-width: 48rem) {
  .Cart__Body {
        width:  calc((100% - (4 * 0.25rem + 4 * 1rem)) / 2);
    }
}

@media (max-width: 30rem) {
  .Cart__Body {
        width: 100%;
    }

  .Checkout__And__Continue__Container {
    display: flex;
    flex-direction: column;
  }
  .Shop__button{
    margin: 0.62rem;
  }
}