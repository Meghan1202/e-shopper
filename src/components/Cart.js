import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './Cart.css';

const Cart = ({ cartItems }) => {
  const totalValue = cartItems.reduce(
    (cartTotal, currentItem) => cartTotal + currentItem.count * currentItem.price, 0,
  );
  return (
    <>
      <table className="Cart__table">
        <thead>
          <tr>
            <th>Item Description</th>
            <th>Unit Price</th>
            <th>Quantity</th>
            <th>Sub Total</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Fruits and Vegitables</td>
          </tr>
          {cartItems.map((cartItem) => (
            <tr key={cartItem.id}>
              <td>{cartItem.id}</td>
              <td>{cartItem.productName}</td>
              <td>{cartItem.count}</td>
              <td>{cartItem.count * cartItem.price}</td>
            </tr>
          ))}
        </tbody>
      </table>
      <div className="Total__Value">
        <h3>
          TOTAL
          {' '}
          {totalValue}
        </h3>
        <hr />
        <div className="Total__Value__container">
          {totalValue > 0
            ? (
              <Link to="/Checkout">
                {' '}
                <button type="button" id="Checkout__button">CheckOut</button>
              </Link>
            ) : alert('No item in the My Basket')}
        </div>
      </div>
    </>
  );
};

Cart.propTypes = {
  cartItems: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    companyName: PropTypes.string.isRequired,
    productName: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    quantity: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
    imgSrc: PropTypes.string.isRequired,
  })).isRequired,
};

export default Cart;
