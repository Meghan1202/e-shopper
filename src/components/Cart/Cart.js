import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './Cart.css';
import CartTable from '../CartTable/CartTable';
import '../CartTable/CartTable.css';

const Cart = ({ cartItems }) => {
  let totalValue = 0;
  Object.keys(cartItems).forEach((category) => {
    cartItems[category].forEach((item) => {
      totalValue += (item.price * item.count);
    });
  });
  return (
    <>
      <div className="Cart__Body" data-testid="Cart">
        <div className="Cart__Container">
          <CartTable cartItems={cartItems} />
        </div>
        <div className="Checkout__And__Continue__Container">
          <Link to="/">
            <button type="button" className="Shop__button">Continue Shopping</button>
          </Link>

          <div className="Total__Value">
            <div className="Value__Container">
              <h3>
                TOTAL
                {' '}
              </h3>
              <h3>
                Rs.
                {' '}
                {totalValue}
              </h3>
            </div>
            <hr />
            <div className="Total__Value__container">
              <Link to="/checkout">
                {' '}
                <button type="button" className="Checkout__button">CheckOut</button>
              </Link>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};
Cart.propTypes = {
  cartItems: PropTypes.objectOf(PropTypes.arrayOf(Object)).isRequired,
};

export default Cart;
