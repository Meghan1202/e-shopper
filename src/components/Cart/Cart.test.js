import {
  fireEvent, render, screen, act,
} from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Cart from './Cart';

describe(Cart.name, () => {
  const mockCartItems = {
    'Fruits & Vegatables': [
      {
        category: 'Fruits & Vegatables',
        companyName: 'Bigbasket',
        count: 2,
        id: 3,
        imgSrc: 'abc',
        name: 'grapes',
        price: 50,
        stock: 8,
      },
    ],
  };
  it('should match the snapshot', () => {
    const { container } = render(<BrowserRouter><Cart cartItems={mockCartItems} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });
  it('should take to checkout page onClick of button', () => {
    render(<BrowserRouter><Cart cartItems={mockCartItems} /></BrowserRouter>);
    const checkOutButton = screen.getByText('CheckOut');
    act(() => { fireEvent.click(checkOutButton); });
    expect(document.location.href).toBe('http://localhost/checkout');
  });
  it('should render the cart table', () => {
    render(<BrowserRouter><Cart cartItems={mockCartItems} /></BrowserRouter>);
    screen.getByTestId('Cart-Table');
  });
  it('Should display the category name', () => {
    render(<BrowserRouter><Cart cartItems={mockCartItems} /></BrowserRouter>);
    screen.getByText('Bigbasket');
  });
});
