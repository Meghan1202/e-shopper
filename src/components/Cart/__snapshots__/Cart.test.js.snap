// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should match the snapshot 1`] = `
<div>
  <div
    class="Cart__Body"
    data-testid="Cart"
  >
    <div
      class="Cart__Container"
    >
      <div
        class="Cart__Table__Container"
        data-testid="Cart-Table"
      >
        <table
          class="Cart__Table"
        >
          <thead>
            <tr>
              <th
                class="Item__Header"
              >
                Item Description
              </th>
              <th
                class="Item__Header"
              >
                Unit Price
              </th>
              <th
                class="Item__Header"
              >
                Quantity
              </th>
              <th
                class="Item__Header"
              >
                Sub Total
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="Row__Header"
                colspan="4"
              >
                Fruits & Vegatables
              </td>
            </tr>
            <tr>
              <td
                class="Item__Row"
              >
                <span
                  class="Brand__Name"
                >
                  Bigbasket
                </span>
                <br />
                grapes
              </td>
              <td
                class="Item__Row"
              >
                50
              </td>
              <td
                class="Item__Row"
              >
                2
              </td>
              <td
                class="Item__Row"
              >
                100
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="Checkout__And__Continue__Container"
    >
      <a
        href="/"
      >
        <button
          class="Shop__button"
          type="button"
        >
          Continue Shopping
        </button>
      </a>
      <div
        class="Total__Value"
      >
        <div
          class="Value__Container"
        >
          <h3>
            TOTAL
             
          </h3>
          <h3>
            Rs.
             
            100
          </h3>
        </div>
        <hr />
        <div
          class="Total__Value__container"
        >
          <a
            href="/checkout"
          >
             
            <button
              class="Checkout__button"
              type="button"
            >
              CheckOut
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
